src/hierarchical-menu/hierarchical-menu-item.ts

```diff
@@ -43,13 +39,13 @@ export type HierarchicalMenuItem = {
 })
 export class NgAisHierarchicalMenuItem {
   @Input() public lvl: number = 1;
-  @Input() public refine: (string) => void;
-  @Input() public createURL: (string) => string;
+  @Input() public refine: HierarchicalMenuState['refine'];
+  @Input() public createURL: HierarchicalMenuState['createURL'];
```

Comment:
IIRC TypeScript struggles to infer the type with this notation. We could rather create a dedicated type for the value. We can then re-use it at both places. I don't have a strong opinion on that though if it works well for you keep it as is.

Reply:
I found that it [worked pretty well](https://typescript-play.js.org/#code/C4TwDgpgBAglC8UDeAoK6oCcIDMCWAdhAFxQAUAbgIYA2AriVAM7CaEDmAlAgHzOscA3CgC+KFAGMA9gRZZchCAEYopGAG0A5NnxFNAXQRRq9aPD4AmYZJlydii6thb7ew4hMNex2l4DUUJoWmsKuEBZkmkxSALYQnhCanChhERbJQA). I'm curious about in what cases typescript can get lost in type inference. Do you recall an example?

Reply:
Nope I don't unfortunately. Anyway if it's working I'm fine.